public/apps/configuration/utils/password-strength-bar.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiProgress, EuiText } from '@elastic/eui';
+import React from 'react';
+import zxcvbn from 'zxcvbn';
+
+interface PasswordStrengthBarProps {
+  password: string;
+}
+
+export const PasswordStrengthBar = (props: PasswordStrengthBarProps) => {
+  const { password } = props;
+  const passwordStrength = zxcvbn(password);
+  const strength = passwordStrength.score;
```

Comment:
is there a message provided by zxcvbn that we can use directly?

Reply:
The score is simply a value 0-4 indicating the strength of the password. Zxcvbn also provides a suggestions and warnings if applicable, which I have also added - see below in the code and you can also check it out in the video.